app/javascript/components/patient/follow_up_flag/FollowUpFlag.js

```diff
@@ -0,0 +1,277 @@
+                  id="bulk_action_apply_to_household"
+                  label="Apply this change to the entire household that these monitorees are responsible for, if it applies."
+                  checked={this.state.bulk_action_apply_to_household}
+                  onChange={this.handleChange}
+                />
+              </Form.Group>
+            </React.Fragment>
+          )}
+          {this.state.clear_flag && (
+            <Form.Group controlId="clear_flag_reason">
+              <Form.Label>Please include any additional details for clearing the follow-up flag:</Form.Label>
+              <Form.Control as="textarea" rows="2" value={this.state.clear_flag_reason} onChange={this.handleChange} />
+              {this.props.bulk_action && (
+                <p className="mt-3">
+                  If any selected monitorees do not currently have a flag set, their records will not be updated as a result of this action.
+                </p>
+              )}
+            </Form.Group>
+          )}
+        </Modal.Body>
+        <Modal.Footer>
+          <Button id="follow_up_flag_cancel_button" variant="secondary btn-square" onClick={this.props.close}>
+            Cancel
+          </Button>
+          <Button
+            id="follow_up_flag_submit_button"
+            variant="primary btn-square"
+            onClick={this.submit}
+            disabled={(!this.state.clear_flag && this.state.follow_up_reason === '') || this.state.loading || this.state.noMembersSelected}>
+            {this.state.loading && (
+              <React.Fragment>
+                <span className="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>&nbsp;
+              </React.Fragment>
+            )}
+            <span data-for="follow-up-submit" data-tip="">
+              {(this.props.bulk_action || this.state.initial_follow_up_reason === '') && 'Submit'}
+              {!this.props.bulk_action && this.state.initial_follow_up_reason !== '' && !this.state.clear_flag && 'Update'}
+              {!this.props.bulk_action && this.state.clear_flag && 'Clear'}
+            </span>
+            {this.state.noMembersSelected && (
+              <ReactTooltip id="follow-up-submit" multiline={true} place="top" type="dark" effect="solid" className="tooltip-container">
+                <div>Please select at least one household member or change your selection to apply to this monitoree only</div>
+              </ReactTooltip>
+            )}
+            {!this.state.noMembersSelected && !this.state.clear_flag && this.state.follow_up_reason === '' && (
+              <ReactTooltip id="follow-up-submit" multiline={true} place="top" type="dark" effect="solid" className="tooltip-container">
+                <div>Please select a reason for follow-up</div>
+              </ReactTooltip>
+            )}
+          </Button>
+        </Modal.Footer>
+      </React.Fragment>
+    );
+  }
+}
+
+FollowUpFlag.propTypes = {
+  patient: PropTypes.object,
```

Comment:
this and the prop below it are kind of confusing.  I know one is for bulk action and one is not, but is there any way we could make this clearer?